src/components/index/useRequestsTableData.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from "react";
+import { ICell, IRow } from "../table/Table";
+import { DateTime } from "luxon";
+import { oracle } from "@uma/sdk";
+import { StyledLink } from "./RequestsTable.styled";
+// import parseAncillaryData from "helpers/parseAncillaryData";
+import { parseIdentifier } from "helpers/format";
+import { ethers } from "ethers";
+import formatYesNoQueryString from "helpers/formatYesNoQueryString";
+const hc: ICell[] = [
+  {
+    size: "lg",
+    value: "Query",
+    cellClassName: "first-header-cell",
+  },
+  {
+    size: "sm",
+    value: "Timestamp",
+    cellClassName: "other-header-cell",
+  },
+  {
+    size: "sm",
+    value: "Status",
+    cellClassName: "other-header-cell",
+  },
+];
+
+function useRequestsTableData(
+  requests: oracle.types.state.RequestIndexes,
+  requeryData: boolean,
+  setRequeryData: React.Dispatch<React.SetStateAction<boolean>>
+) {
+  const [rows, setRows] = useState<IRow[]>([]);
+  const [headerCells] = useState<ICell[]>(hc);
+
+  useEffect(() => {
```

Comment:
i dont think this use effect or the useStates are necessary, try running this as a pure function which takes in requests and returns `{rows:nextRows, headerCells:hc}`

Reply:
Changed to pure function.